LBBT-Design-Patterns/TaxBand.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Concurrent;
+
+namespace LBBT_Design_Patterns;
+
+/// <summary>
+///   A banded tax strategy where a <paramref name="taxRate"/> will be applied to any amount
+///   between a <paramref name="lowerThreshold"/> and <paramref name="upperThreshold"/> if specified,
+///   otherwise limitless above <paramref name="lowerThreshold"/>
+/// </summary>
+public class TaxBand(double taxRate, int lowerThreshold, int? upperThreshold = null)
+    : ITaxStrategy
+{
+    public void ApplyTax(int price, ConcurrentBag<int> taxCalculationResults)
+    {
+        double totalTax = 0;
+        if (price > upperThreshold)
+        {
+            totalTax += (upperThreshold.Value - lowerThreshold) * taxRate;
+        }
+        else if (price > lowerThreshold)
+        {
+            totalTax += (price - lowerThreshold) * taxRate;
+        }
+
+        // TODO consider rounding behaviour
+        var resultant = (int)Math.Ceiling(totalTax);
+        if (resultant != 0) taxCalculationResults.Add(resultant);
```

Comment:
Simplified the if statements for calculations. Only adding to the collection if not zero as we don't want to be writing to a blocking concurrent collection unnecessarily. Put in a todo to consider rounding behaviour as obscuring some rounding within this method isn't ideal.